packages/server/src/fhir/references.ts

```diff
@@ -79,11 +81,21 @@ export class FhirReferenceValidator<T extends Resource> {
     if (refParts.length !== 2) {
       return;
     }
+    if (!this.root.meta?.project) {
+      return;
+    }
 
     try {
       const systemRepo = getSystemRepo();
-      const existing = await systemRepo.readReference(reference);
-      if (existing.meta?.project && this.root.meta?.project && existing.meta.project !== this.root.meta.project) {
+      const target = await systemRepo.readReference(reference);
```

Comment:
`validateReferences` is only ever called from `repo.ts`, where I believe we always have the `allowedProjects` array loaded?

Could we make `allowedProjects` an extra param to the `validateReferences` method?